packages/core/src/App/Components/Routes/route-prompt-dialog.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect } from 'react';
+import { Dialog } from '@deriv/components';
+import { localize, Localize } from '@deriv/translations';
+import { Prompt, withRouter } from 'react-router-dom';
+
+const RoutePromptDialog = ({ condition, history }) => {
+    const [should_show, setShow] = React.useState(false);
+    const [is_confirmed, setConfirm] = React.useState(false);
+    const [last_location, setLastLocation] = React.useState(null);
+
+    useEffect(() => {
+        if (is_confirmed && last_location) history.push(last_location.pathname);
+    }, [is_confirmed, last_location]);
+
+    const handleBlockedNavigation = next_location => {
+        if (!is_confirmed) {
+            setLastLocation(next_location);
+            setShow(true);
+            return false;
+        }
+
+        return true;
+    };
+
+    const onConfirm = () => {
+        setShow(false);
+        setConfirm(true);
+    };
+
+    const onCancel = () => setShow(false);
+
+    return (
+        <>
+            <Prompt when={condition} message={handleBlockedNavigation} />
```

Comment:
Each `react-router` instance can have only 1 `<Prompt />` instance. Please work this condition into the existing one (or move the existing one to this).